tests/test_kb_dataset.py

```diff
@@ -0,0 +1,350 @@
+                    else:
+                        self.assertIn((ref_id, ref_type), existing_ids)
+
+    def _assert_ids_globally_unique(
+        self, collection, ids_seen: set, ignore_assertion: bool = False
+    ):
+        if isinstance(collection, dict):
+            for k, v in collection.items():
+                if isinstance(v, dict):
+                    self._assert_ids_globally_unique(v, ids_seen)
+                elif isinstance(v, list):
+                    for elem in v:
+                        self._assert_ids_globally_unique(elem, ids_seen)
+                else:
+                    if k == "id":
+                        if not ignore_assertion:
+                            self.assertNotIn(v, ids_seen)
+                        ids_seen.add(v)
+        elif isinstance(collection, list):
+            for elem in collection:
+                self._assert_ids_globally_unique(elem, ids_seen)
+
+    def _get_referenced_ids(self, example):
+        referenced_ids = []
+
+        for event in example["events"]:
+            for argument in event["arguments"]:
+                referenced_ids.append((argument["ref_id"], "event_arg"))
+
+        for coreference in example["coreferences"]:
+            for entity_id in coreference["entity_ids"]:
+                referenced_ids.append((entity_id, "entity"))
+
+        for relation in example["relations"]:
+            referenced_ids.append((relation["arg1_id"], "entity"))
+            referenced_ids.append((relation["arg2_id"], "entity"))
+
+        return referenced_ids
+
+    def _get_existing_referable_ids(self, example):
+        existing_ids = []
+
+        for entity in example["entities"]:
+            existing_ids.append((entity["id"], "entity"))
+
+        for event in example["events"]:
+            existing_ids.append((event["id"], "event"))
+
+        return existing_ids
+
+    def test_passages_offsets(self):
+        """
+        Verify that the passages offsets are correct,
+        i.e.: passage text == text extracted via the passage offsets
+        """
+
+        for split in self.dataset_bigbio:
+
+            if "passages" in self.dataset_bigbio[split].features:
+
+                for example in self.dataset_bigbio[split]:
+
+                    example_text = _get_example_text(example)
+
+                    for passage in example["passages"]:
+
+                        text = passage["text"]
+                        offsets = passage["offsets"]
+
+                        with self.subTest(
+                            "Offsets in passages must have only one element",
+                            offsets=offsets,
+                        ):
+                            self.assertEqual(len(offsets), 1)
+
+                        with self.subTest(
+                            "Text in passages must be an Iterable",
+                            text=text,
+                        ):
+                            self.assertIsInstance(text, Iterable)
```

Comment:
a string is also an iterable :) but I think this suite of tests will catch all but the most pathological cases and will be a great tool for contributors

Reply:
> a string is also an iterable :)

😅 brain fart